src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -160,7 +156,7 @@ internal partial class DefiniteAssignmentPass : LocalDataFlowPass<
                    trackUnassignments)
         {
             this.initiallyAssignedVariables = null;
-            _sourceAssembly = ((object)member == null) ? null : (SourceAssemblySymbol)member.ContainingAssembly;
+            _sourceAssembly = member?.ContainingAssembly as SourceAssemblySymbol;
```

Comment:
>member?.ContainingAssembly as SourceAssemblySymbol [](http://example.com/codeflow?start=30&length=50)

It feels like we might want to strengthen the logic here to get ```ContainingAssembly``` only from a symbol that is actually a member. Otherwise, we might still get a ```SourceAssemblySymbol``` that doesn't match the context and start changing its state based on the analysis that we perform.